### PR TITLE
Gradle updates and misc code cleanup

### PR DESCRIPTION
**Pull request by TacoTheDank**
### 1st commit
- Update gradle wrapper to 6.4.1 using git bash

### 2nd commit
- Update Sdk and buildtools
- Update dependencies to latest support

### 3rd commit
- Migrate to AndroidX

### 4th commit
- Add an .editorconfig
- Adjust the codestyle to the .editorconfig

### 5th commit
- Automatic code cleanup by Android Studio

### 6th commit
- Fix some typos
- Tiny code cleanup

### 7th commit
- Fix links

### 8th commit
- Update Android Studio 3.6.3 -> 4.0.1
- Corrected a few things

### CODE REVIEW

**Comment by phikal** [comment]
Sorry for answering so late, I totally missed this push request. I'll review it as soon as possible!

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+# indent_size = 4
```
Comment (phikal) [inline .editorconfig:11]: Could this comment be removed? Or is there a point to it?

File: README.md [hunk 2]
```diff
@@ -1,17 +1,21 @@
+[<img src="https://fdroid.gitlab.io/artwork/badge/get-it-on.png"
+    alt="Get it on F-Droid"
+    height="80">](https://f-droid.org/packages/com.phikal.regex/)
```
Comment (phikal) [inline README.md:3]: I'm really not a fan of badges in README's -- they make it harder to read, because of the obfuscated HTML.

File: README.md [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
 
 [regex]: https://en.wikipedia.org/wiki/Regular_expression
-[website]: https://zge.us.to/proj/ReGeX/
-[fdroid]: https://en.wikipedia.org/wiki/Regular_expression
-[github]: https://github.com/phikal/ReGeX/
-[license]: ./LICENSE
+[website]: https://github.com/phikal/ReGeX/wiki
+[fdroid]: https://f-droid.org/packages/com.phikal.regex/
+[github]: https://github.com/phikal/ReGeX
+[license]: https://www.gnu.org/licenses/gpl-3.0.en.html
```
Comment (phikal) [inline README.md:21]: What's the rationale behind this change? GitHub already links to the README, and when viewed locally, one doesn't have to open a browser to read the license.

File: app/build.gradle [hunk 4]
```diff
@@ -1,33 +1,34 @@
...
-        targetSdkVersion 24
+        targetSdkVersion 28
         versionCode 5
         versionName "1.3.1"
     }
+
     buildTypes {
         release {
             minifyEnabled true
-            shrinkResources true
```
Comment (phikal) [inline app/build.gradle:18]: Is there a downside to this I don't know of?

**Comment by phikal** [comment]
The rest is ok, seems like it's mostly automatic updates, right? And the spelling mistakes should have been taken care of a lot earlier, so thanks for that too! Due to the resources required by android studio, I was until recently not able to maintain the app properly, so your fixes should help me clean up the code base.

File: .editorconfig (hunk 1 above)
Comment (TacoTheDank) [reply .editorconfig:11]: It defines the codestyle used throughout the project. It's needed to make local clones use the correct format (otherwise you could get PRs using tabs, and other such things).

File: README.md (hunk 2 above)
Comment (TacoTheDank) [reply README.md:3]: Oh, ok.

File: README.md (hunk 3 above)
Comment (TacoTheDank) [reply README.md:21]: It's just me trying to make the links as correct as possible. It isn't really necessary. I'll change it back if you want.

File: app/build.gradle (hunk 4 above)
Comment (TacoTheDank) [reply app/build.gradle:18]: Nope, I just find that it makes debugging particularly difficult if a crash occurs (minify doesn't help, either, but that's fine).

File: .editorconfig (hunk 1 above)
Comment (phikal) [reply .editorconfig:11]: I'm vaguely familiar with Editorconfig, it's just that lines beginning with `#` [are comments](https://editorconfig-specification.readthedocs.io/en/latest/#file-format), and I didn't see the reason to leave a line commented when it could just as well be uncommented or removed.

File: README.md (hunk 3 above)
Comment (phikal) [reply README.md]: Markdown-wise, they are both equally valid as links. But thinking about it, I guess it's not a big deal, so you can also leave it as it is.

File: app/build.gradle (hunk 4 above)
Comment (phikal) [reply app/build.gradle:18]: But won't it only be shrunk in release builds? It's unusual for people to use them for debugging. I've never really understood Gradle, so forgive my ignorance.

Comment (TacoTheDank) [reply app/build.gradle:18]: It's ok, you're right. I can't actually remember why I removed this haha.

File: .editorconfig (hunk 1 above)
Comment (TacoTheDank) [reply .editorconfig:11]: Oh, you were talking about that. Yeah, that can be removed.

**Comment by TacoTheDank** [comment]
I fixed the .editorconfig and shrinkResources in the Android Studio commit.

**Comment by phikal** [comment]
Looks good then. Thanks a lot for doing this!
